source/client/client_worker_impl.cc

```diff
@@ -46,23 +50,28 @@ void ClientWorkerImpl::work() {
   benchmark_client_->setMeasureLatencies(true);
   sequencer_->start();
   sequencer_->waitForCompletion();
-  benchmark_client_->terminate();
-  success_ = true;
-  dispatcher_->exit();
   // Save a final snapshot of the worker-specific counter accumulations before
   // we exit the thread.
   for (const auto& stat : store_.counters()) {
     // First, we strip the cluster prefix
     std::string stat_name = std::string(absl::StripPrefix(stat->name(), "cluster."));
+    stat_name = std::string(absl::StripPrefix(stat_name, "worker."));
     // Second, we strip our own prefix if it's there, else we skip.
-    const std::string worker_prefix = fmt::format("worker.{}.", worker_number_);
+    const std::string worker_prefix = fmt::format("{}.", worker_number_);
     if (stat->value() && absl::StartsWith(stat_name, worker_prefix)) {
       thread_local_counter_values_[std::string(absl::StripPrefix(stat_name, worker_prefix))] =
           stat->value();
     }
   }
 }
 
+void ClientWorkerImpl::shutdownThread() {
+  // Terminate will shut down the pool and run the dispatcher, which may increment certain stats
+  // like the number of connections destroyed. We do that here, so we don't pollute the reported
```

Comment:
Here vs. where? This is not clear from the comment.

Reply:
Modified comment and moved it above, let me know if that looks better.